ethpm/utils/manifest_validation.py

```diff
@@ -54,11 +54,11 @@ def validate_manifest_deployments(manifest: Dict[str, Any]) -> None:
     if set(("contract_types", "deployments")).issubset(manifest):
         all_contract_types = list(manifest["contract_types"].keys())
         all_deployments = list(manifest["deployments"].values())
-        all_deployment_names = set(
-            itertools.chain.from_iterable(deployment for deployment in all_deployments)
-        )
+        all_deployment_contract_types = [
+            cytoolz.first(dep.values())["contract_type"] for dep in all_deployments
+        ]
```

Comment:
This was incorrect as previously we were using the deployment dictionary key (contract_name) to compare against `all_contract_types` instead of the 'contract_type' key inside the deployment.
However, i'm not sure if this approach is too complex / unreadable? Open to any suggestions on improving the readability

Reply:
I'm actually not sure this approach is correct.  The `cytoolz.first` call seems like it would truncate the deployment contracts for a manifest that has multiple deployments.  It seems like this should be

```
all_deployment_names = set(
    deployment['contract_type'] for chain_deployments in all_deployments
    for deployment in chain_deployments.values()
)
```

This is probably worth writing a simple utility `extract_contract_types_from_deployments` so that we can write some quick tests for it.